pkg/resource/generator.go

```diff
@@ -285,5 +287,25 @@ func (g *Generator) Remove(cr *imageregistryv1.Config, modified *bool) error {
 		*modified = true
 	}
 
+	driver, err := storage.NewDriver(cr.ObjectMeta.Name, cr.ObjectMeta.Namespace, &cr.Status.Storage)
+	if err != nil {
+		return err
+	}
+
+	var derr error
+	err = wait.PollImmediate(1*time.Second, 5*time.Minute, func() (stop bool, err error) {
+		if retriable, derr := driver.RemoveStorage(cr, modified); derr != nil {
+			if retriable {
+				return false, nil
+			} else {
+				return true, err
```

Comment:
derr?

Reply:
yeah this logic is a little borked now.  returning derr is probably the simplest fix.  As it stands now you're risking returning a nil error when you actually got a fatal error.

Reply:
yep, that was a typo, updated in current push